### DISCUSSION TITLE
Where i can find _ba<lib name>.py stub files for modding?

### DISCUSSION THREAD

**Discussion by TheNotEasy** [efroemling/ballistica#592] [Q&A] [answered]
Example: _bauiv1.py

I want to make mods already with refactoring. Should I wait for release?

**Comment by efroemling** [comment, marked as answer]
As of 1.7.20, these 'dummy-module' files are generated as part of builds so that they get automatically regenerated if any of the underlying c++ stuff changes. So if you can get to the point where you can run `make mypy` or anything like that, then you should wind up with dummy modules generated as a side effect in `build/dummymodules`. You can also just do `make dummymodules` to ask for them directly.

Though this gets me wondering if a lot of modders don't have full environments set up and thus aren't able to easily generate them that way. Would it be worth making a static copy of them available somehow without needing to build?
